packages/salesforcedx-vscode-apex/src/commands/apexActionController.ts

```diff
@@ -132,4 +130,127 @@ export class ApexActionController {
       .join('\n');
     return stringify(parse(theDoc));
   }
+
+  private saveOasAsErsMetadata = async (oasSpec: string, fullPath: string): Promise<void> => {
+    // Replace the schema section in the ESR file if it already exists
+    let existingContent;
+    if (fs.existsSync(fullPath)) {
+      existingContent = fs.readFileSync(fullPath, 'utf8');
+    }
+    const namedCredential = await this.showNamedCredentialsQuickPick();
+
+    const updatedContent = existingContent
```

Comment:
@CristiCanizales Would this be better handled using a XML document, rather than doing all of this using string manipulation?